### PR TITLE
feat(CD): add action to trigger deploy pipeline on push on main branch

### PR DESCRIPTION
**Pull request by pietro-tota**
#### List of Changes
Add github action that check for the presence of well-formed tags that will be used to trigger semantic versioning parameters for deploy module once PR is merge on main branch.
Those tags are:

| Tag                | Semantic versioning scope | Meaning                                                           |
|--------------------|---------------------------|-------------------------------------------------------------------|
| patch              | Application version       | Patch-bump application version into pom.xml and Chart app version |
| minor              | Application version       | Minor-bump application version into pom.xml and Chart app version |
| major              | Application version       | Major-bump application version into pom.xml and Chart app version |
| ignore-for-release | Application version       | Ignore application version bump                                   |
| chart-patch        | Chart version             | Patch-bump Chart version                                          |
| chart-minor        | Chart version             | Minor-bump Chart version                                          |
| chart-major        | Chart version             | Major-bump Chart version                                          |
| skip-release       | Any                       | The release will be skipped altogether                            |

For the check to be successfully passed only one of the `Application version` labels and only ones of
the `Chart version` labels must be contemporary present for a given PR or the `skip-release` for skipping release step

Those tags are then read back once the PR is merged against main branch to value azure devops deploy pipeline

#### Motivation and Context

Those modifications are needed to enable CI with UAT deploy once a PR is merged into main branch

#### Types of changes

Github action run on repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          echo "SKIP_RELEASE=${{ steps.configure_pipeline_parameters.outputs.SKIP_RELEASE }}"
+          echo "UAT_SKIP_BLUE_DEPLOYMENT=${{ steps.configure_pipeline_parameters.outputs.UAT_SKIP_BLUE_DEPLOYMENT }}"
+          echo "PROD_SKIP_BLUE_DEPLOYMENT=${{ steps.configure_pipeline_parameters.outputs.PROD_SKIP_BLUE_DEPLOYMENT }}"
+          echo "SKIP_BUILD=${{ steps.configure_pipeline_parameters.outputs.SKIP_BUILD }}"
+        shell: bash
+      - name: Azure Pipelines Action
+        uses: jacopocarlini/azure-pipelines@v1.3
+        with:
+          azure-devops-project-url: https://dev.azure.com/pagopaspa/pagoPA-projects
+          azure-pipeline-name: 'pagopa-ecommerce-transactions-scheduler-service.deploy'
```
Comment (pietro-tota) [inline .github/workflows/deploy.yml:72]:
```suggestion
          azure-pipeline-name: 'pagopa-ecommerce-payment-requests-service.deploy'
```

**Review by Fernando-Granato — APPROVED** [review]

File: .github/actions/check-pr-semver-labels/action.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              commit_sha: context.sha,
+            });
+            if (!mergeCommit) {
+              throw `Could not find commit for sha ${context.sha}`;
+            }
+            const commitMessage = mergeCommit.data.message;
+            core.info(`Commit message: ${commitMessage}`);
+            const extractedPrNumber = commitMessage.match('\\(#(\\d{1,9})\\)');
```
Comment (giovanniberti) [inline .github/actions/check-pr-semver-labels/action.yml:49]: Doesn't this match only PRs with number < 10?

Comment (pietro-tota) [reply .github/actions/check-pr-semver-labels/action.yml:49]: No, this is a greedy quantifier, so it match all pr from #1 to #999999999

Comment (giovanniberti) [reply .github/actions/check-pr-semver-labels/action.yml:49]: Oh right, I didn't see `{1,9}` correctly

Comment (giovanniberti) [reply .github/actions/check-pr-semver-labels/action.yml:49]: Btw, you can use regex literals to improve readability a bit:
```suggestion
            const extractedPrNumber = commitMessage.match(/\(#(\d{1,9})\)/); 
```

File: .github/actions/check-pr-semver-labels/action.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+              commit_sha: context.sha,
+            });
+            if (!mergeCommit) {
+              throw `Could not find commit for sha ${context.sha}`;
+            }
+            const commitMessage = mergeCommit.data.message;
+            core.info(`Commit message: ${commitMessage}`);
+            const extractedPrNumber = commitMessage.match('\\(#(\\d{1,9})\\)'); 
+            if(extractedPrNumber ==null || extractedPrNumber.length <1){
+               throw `Could not extract PR number from commit message ${commitMessage}`;
```
Comment (giovanniberti) [inline .github/actions/check-pr-semver-labels/action.yml:51]:
```suggestion
            if(extractedPrNumber == null || extractedPrNumber.length < 1){
               throw `Could not extract PR number from commit message ${commitMessage}`;
```

**Review by giovanniberti — APPROVED** [review]
Well done! 🚀
Left a couple minor comments 🙂

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**